lstchain/scripts/lstchain_find_pedestals.py

```diff
@@ -0,0 +1,186 @@
+
+parser.add_argument('-d', '--input-dir', dest='srcdir',
+                    type=Path, default='./',
+                    help='path to the directory of the DL1 files'
+                    )
+
+parser.add_argument('output_dir', metavar='output-dir', type=Path,
+                    help='Path where to store the output hdf5 files'
+                    )
+
+
+def main():
+
+    args = parser.parse_args()
+
+    files = sorted(args.srcdir.glob('dl1_LST-1.Run?????.????.h5'))
+    if not files:
+        raise IOError("No input dl1 files found")
+
+    output_dir = args.output_dir.absolute()
+    output_dir.mkdir(exist_ok=True, parents=True)
+
+    for file in files:
+        run_info = parse_dl1_filename(file)
+        run_number, subrun_index = run_info.run, run_info.subrun
+
+        # Approximate interleaved pedestal frequency in Hz. This changed once (so far) in the life of LST.
+        # We just use the run number to decide which frequency we should expect:
+        approximate_frequency = 50
+        if run_number > 2708:
+            approximate_frequency = 100
+
+        table = read_table(file, '/dl1/event/telescope/parameters/LST_LSTCam')
+        all_times = np.array(table['dragon_time'])
+        # probable interleaved flat-field events (just in case tag is
+        # faulty!):
+        ffmask = ((table['intensity'] > 3e4) &
+                  (table['concentration_pixel'] < 0.005))
+
+        pedmask = np.zeros(len(table), dtype=bool)
+        pedmask[~ffmask] = find_pedestals(all_times[~ffmask],
+                                          approximate_frequency)
+
+        # Now remove the 10 brightest events (might be cosmics accidentally
+        # falling in the time windows determined by time_pedestals). The
+        # expected value of cosmics is smaller that, so we are probably
+        # removing some pedestals, but it does not harm.
+
+        intensity = np.array(table['intensity'])
+        # set nans to 0, to avoid problems with sorting:
+        intensity[np.isnan(intensity)] = 0
+
+        decreasing_intensity_ordered_indices = np.flip(np.argsort(intensity))
+        pedmask2 = pedmask[decreasing_intensity_ordered_indices]
+        # indices of likely pedestals, ordered from brightest to dimmest:
+        remove_indices = decreasing_intensity_ordered_indices[pedmask2]
+        pedmask[remove_indices[:10]] = False
+
+        event_id = table['event_id']
+        print(file)
+        print('  Rate of identified pedestals: {:.3f}'.format(pedmask.sum() /
+                                                              (all_times[-1] -
+                                                               all_times[0])),
+                                                              'Hz')
+        if pedmask.sum() == 0:
+            continue
+
+        print('  Maximum intensity:', np.nanmax(intensity[pedmask]), 'pe')
+
+        df = pd.DataFrame({'event_id': event_id[pedmask]})
```

Comment:
Please also add the `obs_id` column, so we can check if the correct file was given.

Reply:
The name of the output file (one per subrun) uses the same pattern as the input file, so unless there is already a mismatch in the DL1  file between obs_id and Run in the file name.